app/uk/gov/hmrc/exports/services/mapping/CachingMappingHelper.scala

```diff
@@ -17,24 +17,34 @@
 package uk.gov.hmrc.exports.services.mapping
 
 import uk.gov.hmrc.exports.models.declaration.IdentificationTypeCodes.{CUSCode, CombinedNomenclatureCode, NationalAdditionalCode, TARICAdditionalCode}
-import uk.gov.hmrc.exports.models.declaration.{CommodityMeasure, ExportItem}
+import uk.gov.hmrc.exports.models.declaration.{CommodityDetails, CommodityMeasure, ExportItem}
 import uk.gov.hmrc.wco.dec._
 
 class CachingMappingHelper {
   val defaultMeasureCode = "KGM"
 
-  def commodityFromExportItem(exportItem: ExportItem): Commodity = {
-    def removeCarriageReturns(description: String): String = description.replaceAll("(\\r\\n)|\\r|\\n", " ")
+  def commodityFromExportItem(exportItem: ExportItem): Option[Commodity] = {
+    def removeCarriageReturns(description: Option[String]): Option[String] = description.map(_.replaceAll("(\\r\\n)|\\r|\\n", " "))
 
-    Commodity(
-      description = exportItem.commodityDetails.map(commodityDetails => removeCarriageReturns((commodityDetails.descriptionOfGoods))),
-      classifications = getClassifications(
-        exportItem.commodityDetails.flatMap(_.combinedNomenclatureCode),
-        exportItem.cusCode.flatMap(_.cusCode),
-        exportItem.nactCodes.map(_.map(_.nactCode)).getOrElse(List.empty),
-        exportItem.taricCodes.map(_.map(_.taricCode)).getOrElse(List.empty)
-      ),
-      dangerousGoods = exportItem.dangerousGoodsCode.flatMap(_.dangerousGoodsCode).map(code => Seq(DangerousGoods(Some(code)))).getOrElse(Seq.empty)
+    exportItem.commodityDetails.flatMap(
+      commodityDetails =>
+        commodityDetails match {
+          case CommodityDetails(None, None) => None
```

Comment:
What if CommodityDetails is (None, None) but there are/is a dangerousGoodsCode on the item.  It will be lost now won't it?

I would only return None (CommodityDetails) iff
- no descriptionOfGoods
- classifications is empty
- no dangerousGoodsCode

Reply:
Similar - you are discarding the CUS, NACT and TARIC codes at the moment

Reply:
:man_facepalming: 
Thank you once again. 
I have used a private method to check if the object is "empty", instead of adding it to the wco-dec model. 
We will have to remove the wco-dec from this project, although it depends on it. :shrug: